modin/core/storage_formats/pandas/parsers.py

```diff
@@ -663,43 +663,66 @@ class ParquetFileToRead(NamedTuple):
 
 @doc(_doc_pandas_parser_class, data_type="PARQUET data")
 class PandasParquetParser(PandasParser):
+    @staticmethod
+    def _read_row_group_chunk(
+        f, row_group_start, row_group_end, columns, engine
+    ):  # noqa: GL08
+        if engine == "pyarrow":
+            from pyarrow.parquet import ParquetFile
+
+            return (
+                ParquetFile(f)
+                .read_row_groups(
+                    list(
+                        range(
+                            row_group_start,
+                            row_group_end,
+                        )
+                    ),
+                    columns=columns,
+                    use_pandas_metadata=True,
+                )
+                .to_pandas()
+            )
+        elif engine == "fastparquet":
+            from fastparquet import ParquetFile
+
+            return ParquetFile(f)[row_group_start:row_group_end].to_pandas(
```

Comment:
won't this read the entire dataset? or is `ParquetFile` lazy?

Reply:
It won't read the entire dataset. I looked through the code base to confirm this as well. `ParquetFile` will initialize some metadata and then the slice for the row groups just sets some stuff internally and will actually be read in during the `to_pandas` call.